packages/plugin-telegram-api/src/receiveMessage.ts

```diff
@@ -0,0 +1,139 @@
+import { Message } from 'telegraf/typings/core/types/typegram';
+import { Chats, Customers, Integrations, Messages } from './models';
+import { sendContactsMessage, sendInboxMessage } from './messageBroker';
+import { Context } from 'telegraf';
+
+const getOrCreateCustomer = async (
+  message: Message.TextMessage,
+  integration
+) => {
+  const from = message.from?.id;
+
+  const prev = await Customers.findOne({ telegramId: from });
+
+  let customerId;
+
+  if (prev) {
+    return prev.contactsId;
+  }
+
+  const customer = await sendContactsMessage({
+    subdomain: 'os',
+    action: 'customers.findOne',
+    data: {
+      telegramId: from
+    },
+    isRPC: true
+  });
+
+  if (customer) {
+    customerId = customer._id;
+  } else {
+    const apiCustomerResponse = await sendContactsMessage({
+      subdomain: 'os',
+      action: 'customers.createCustomer',
+      data: {
+        firstName: message.from?.first_name,
+        lastName: message.from?.last_name,
+        telegramId: from
+      },
+      isRPC: true
+    });
+
+    customerId = apiCustomerResponse._id;
+  }
+
+  await Customers.create({
+    inboxIntegrationId: integration._id,
+    contactsId: customerId,
+    telegramId: from
+  });
+
+  return customerId;
+};
+
+export const receiveMessage = (accountId: string) => async (ctx: Context) => {
+  // @ts-ignore
+  const message = ctx.update.message as Message.TextMessage;
+
+  // Don't listen for bot messages
+  if (message.from?.is_bot) return;
+
+  // Ignore private messages
+  if (message.chat.type === 'private') return;
```

Comment:
@bpedersen why ignore private messages?

Reply:
Ok, wasn't sure on how you wanted to track it. I can make that change.

Reply:
The way that it is configured now is that the integration is on the per chatroom level. So when setting up the integration you have to specify which channel it is creating a conversation in. Would we like to make a integration per bot instead?